src/Lykke.Service.Zcash.Api.Services/HealthService.cs

```diff
@@ -7,13 +9,30 @@ namespace Lykke.Service.Zcash.Api.Services
     // NOTE: See https://lykkex.atlassian.net/wiki/spaces/LKEWALLET/pages/35755585/Add+your+app+to+Monitoring
     public class HealthService : IHealthService
     {
-        public string GetHealthViolationMessage()
+        private readonly IBlockchainReader _blockchainReader;
+
+        public HealthService(IBlockchainReader blockchainReader)
+        {
+            _blockchainReader = blockchainReader;
+        }
+
+        public async Task<string> GetHealthViolationMessage()
         {
             // TODO: Check gathered health statistics, and return appropriate health violation message, or NULL if service hasn't critical errors
+
+            try
+            {
+                var info = await _blockchainReader.GetInfoAsync();
```

Comment:
info - unused variable

Reply:
Isalive should not make synchronous IO. Move it to the TimerTrigger

Reply:
Fixed. Use gathering issues while performing RPC requests instead of direct check.